src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -869,6 +864,36 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
             return retVal;
         }
 
+        /// <summary>
+        /// Prepare headers in a format that can be fed to curl API
+        /// </summary>
+        /// <param name="headers"></param>
+        /// <returns>List of headers</returns>
+        private static List<string> PrepareRequestHeaders(params HttpHeaders[] headers)
+        {
+            List<string> allHeaders = new List<string>(headers.Length);
+
+            foreach (var multiHeaders in headers)
+            {
+                if (multiHeaders != null)
+                {
+                    foreach (var header in multiHeaders)
+                    {
+                        string headerStr = header.Key + ":";
+
+                        foreach (var headerValue in header.Value)
+                        {
+                            headerStr += " " + headerValue;
```

Comment:
FYI: HTTP allows multiple headers with the same (which should be interchangeable with comma separated values), but HttpClient's representation of the user-agent header is wrong. Instead of multiple values separated by spaces, it should be a single value that may contain spaces (like an authorization header value). Merging multiple user-agent values with space separators is the appropriate workaround. You shouldn't have this problem with other headers.